### PR TITLE
Improved key state and added hotkeys for closing/temporary disable

### PR DESCRIPTION
**Pull request by asamy**
This should help a little, I am not sure if the hotkeys are actually suitably
aligned...

Following hotkeys are new:
    LCTRL+LALT
        +A => toggle
        +C => shut down

### CODE REVIEW

**Comment by taviso** [comment]
Thanks! I like this, but I'm wondering if it can be made easier to hack if someone wants different hotkeys, something like this at the top with an array of keycodes and callbacks:

```
static const FARPROC HotKeyCallbacks[256] = {
 ['A'] = ToggleCorners,
 ['C'] = ExitCorners,
 ['G'] = GrowCornerZone,
 ['S'] = ShrinkCornerZone,
 ['Z'] = SwitchToNextCorner,
// etc
}
```
Then disabling/adding/changing hotkeys and recompiling is easier?

**Comment by asamy** [comment]
Looks good now?
Maybe we should also remove the mouse state checks in corner thread?

File: hotcorner.c [hunk 1]
```diff
@@ -35,12 +35,31 @@ static const INPUT kCornerInput[] = {
     { .type = INPUT_KEYBOARD, .ki = { .wVk = VK_LWIN, .dwFlags = KEYEVENTF_KEYUP }},
 };
 
+// Callbacks for hotkeys.  Defined below.
+static void ToggleCorners(void);
+static void ExitCorners(void);
+
+static const FARPROC HotKeyCallbacks[] = {
```
Comment (taviso) [inline hotcorner.c]: Specify the size as [256] so that the undefined entries are NULL.

File: hotcorner.c [hunk 2]
```diff
@@ -162,10 +181,13 @@ static LRESULT CALLBACK KeyHookCallback(int nCode, WPARAM wParam, LPARAM lParam)
 
     KeyState[vkCode] = wParam == WM_SYSKEYDOWN || wParam == WM_KEYDOWN;
     if (KeyState[VK_LCONTROL] && KeyState[VK_LMENU]) { /* LCTRL + LALT  */
-        if (KeyState[0x41])     /* +A => toggle  */
-            disabled = !disabled;
-        else if (KeyState[0x43])    /* +C => die  */
-            TerminateProcess(GetCurrentProcess(), 0);
+        /* Loop up to hotkeycallbacks size not key state size...  */
+        for (size_t k = 0; k < sizeof(HotKeyCallbacks) / sizeof(HotKeyCallbacks[0]); ++k) {
```
Comment (taviso) [inline hotcorner.c]: I think this should be _countof(HotKeyCallbacks)

**Comment by taviso** [comment]
I like it, merged. Thanks!

**Comment by taviso** [comment]
Oops, it doesn't work with my version of visual studio, let me try to fix it up.

**Comment by asamy** [comment]
That's odd, I used VS 2015 compiler.

**Comment by taviso** [comment]
Ah, I had some debugging flags and it didn't like the FARPROC cast, I just rearranged it a bit.

**Pull request merged**